pkg/cmd/resource/operation.go

```diff
@@ -51,7 +52,22 @@ func (oc *OperationCmd) runOperationCmd(cmd *cobra.Command, args []string) error
 		// only include fields explicitly set by the user to avoid conflicts between e.g. account_balance, balance
 		if oc.Cmd.Flags().Changed(stringProp) {
 			paramName := strings.ReplaceAll(stringProp, "-", "_")
-			flagParams = append(flagParams, fmt.Sprintf("%s=%s", paramName, *stringVal))
+			if strings.Contains(paramName, ".") {
+				fullParam := constructParamFromDot(paramName)
+				flagParams = append(flagParams, fmt.Sprintf("%s=%s", fullParam, *stringVal))
+			} else {
+				flagParams = append(flagParams, fmt.Sprintf("%s=%s", paramName, *stringVal))
+			}
+		}
+	}
+
+	for arrayProp, arrayVal := range oc.arrayFlags {
+		// only include fields explicitly set by the user to avoid conflicts between e.g. account_balance, balance
```

Comment:
Good idea to do this! Can we have a test case for it?

Reply:
This should be covered by `TestRunOperationCmd`. That has array data passed into it that it validates is correctly parsed and handled